src/app/containers/PageHandlers/withHashChangeHandler/index.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { memo, useEffect } from 'react';
+import path from 'ramda/src/path';
+import pipe from 'ramda/src/pipe';
+
+import isHashChange from './isHashChange';
+
+const useHashChangeHandler = hash => {
+  useEffect(() => {
+    if (hash) {
+      const element = document.querySelector(hash);
+
+      element?.scrollIntoView();
+      element?.focus();
+    }
+  }, [hash]);
+};
+
+const getHashProp = path(['location', 'hash']);
+
+const withHashChangeHandler = Component => props => {
+  const hash = getHashProp(props);
+
+  useHashChangeHandler(hash);
+
+  return <Component {...props} />;
+};
+
+const withMemoization = Component => memo(Component, isHashChange);
+
+export default pipe(withMemoization, withHashChangeHandler);
```

Comment:
Is the memoization the technique used to prevent the re-render? The react docs [here](https://reactjs.org/docs/react-api.html#reactmemo) discourage that approach

Another blog post on this: https://dmitripavlutin.com/use-react-memo-wisely/#5-reactmemo-is-a-performance-hint

I'm not sure how else you would do this though and:
- It is probably fairly likely to prevent a re-render (though not guaranteed)
- It doesn't break the app if a re-render does occur, it is just undesirable.